modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -225,6 +225,50 @@ TEST_P(Test_ONNX_layers, Gather)
     testONNXModels("gather", npy, 0, 0, false, false);
 }
 
+TEST_P(Test_ONNX_layers, RAFT)
+{
+    std::string weight_path = _tf("models/optical_flow_estimation_raft_2023aug.onnx", false);
+    std::string img0_path = findDataFile(std::string("gpu/opticalflow/frame0.png"));
+    std::string img1_path = findDataFile(std::string("gpu/opticalflow/frame1.png"));
+
+    Size target_size{480, 360};
```

Comment:
@fengyuentau, can we reduce input resolution as minimul as possible to cover required checks for a new layer? `240x180` or lower.

Reply:
It may be feasible, but the model may be sensitive to the input shape. The intention was we use the model in opencv_zoo for accuracy or performance testing. If the model is shape-sensitive in the end, we have to replace the model in the zoo with one with a smaller input shape. That could be some trouble.

Reply:
Can you please reduce precision to FP16? Some changes in `blobFromNPY` will be required.

Reply:
I will handle this problem after we can run this model. Currently we are facing some other issues running this model.

Reply:
How about we just check the smaller output? This model has two outputs, one of which has shape [1, 2, 45, 60] and file size 21KB, the other of which has shape [1, 2, 360, 480] and file size 1.32MB. The bigger output is calculated from the smaller output. I think it also makes sense if we just check the smaller one.

Reply:
I found that even though problems of `GatherElements` have been fixed, the inference results of RAFT between ours and onnxruntime are different. Need an investigation into this problem.

Reply:
> I found that even though problems of `GatherElements` have been fixed, the inference results of RAFT between ours and onnxruntime are different. Need an investigation into this problem.

I found we have different results on the InstanceNorm layer from ONNX Runtime. Let me fix it.

Reply:
@fengyuentau, thanks for the investigation. If the biggest output is computed from the smallest one, it's a good ides to check the smallest only as it will cover most critical layers.

Reply:
With https://github.com/opencv/opencv/pull/24378, the difference is reduced to

```
[ RUN      ] Test_ONNX_layers.RAFT/0, where GetParam() = OCV/CPU
/Users/fengyuantao/Workspace/fytao/projects/gsoc_optical_flow/opencv/modules/dnn/test/test_common.impl.hpp:76: Failure
Expected: (normL1) <= (l1), actual: 0.00010332 vs 1e-05
  |ref| = 410.04229736328125
/Users/fengyuantao/Workspace/fytao/projects/gsoc_optical_flow/opencv/modules/dnn/test/test_common.impl.hpp:79: Failure
Expected: (normInf) <= (lInf), actual: 0.00147033 vs 0.0001
  |ref| = 410.04229736328125
[  FAILED  ] Test_ONNX_layers.RAFT/0, where GetParam() = OCV/CPU (1738 ms)
[----------] 1 test from Test_ONNX_layers (1738 ms total)
```

Still need a further investigation.

Reply:
Seems good, as for me. l1=2e-4 and lInf=2e-3

Reply:
Found another bug on the element-wise layer, see https://github.com/opencv/opencv/issues/24385.

Reply:
Finally dnn can achieve the same accuracy with https://github.com/opencv/opencv/pull/24386 (greater fix), with or without #24378 (instancenorm).

Reply:
This RAFT model takes nearly 2s for a single inference on my mac m1. I think we still need to replace with a smaller input scale version.

Reply:
Tested with a smaller input one but the time is almost the same. Lets keep what we have. Also I found that we have almost the same inference time compared to ONNX Runtime.

Reply:
I suggest to add a tag to this test `applyTestTag(CV_TEST_TAG_LONG, CV_TEST_TAG_DEBUG_VERYLONG, CV_TEST_TAG_MEMORY_2GB);` to skip as long. PR is ready and might be merged.

Reply:
> I suggest to add a tag to this test `applyTestTag(CV_TEST_TAG_LONG, CV_TEST_TAG_DEBUG_VERYLONG, CV_TEST_TAG_MEMORY_2GB);` to skip as long. PR is ready and might be merged.

Thank you for the tips. I almost forgot these flags!